source/common/protobuf/utility.h

```diff
@@ -177,6 +178,51 @@ class MessageUtil {
   static void loadFromYaml(const std::string& yaml, Protobuf::Message& message);
   static void loadFromFile(const std::string& path, Protobuf::Message& message);
 
+  static void checkForDeprecation(const Protobuf::Message& message, bool fatal) {
+    const Protobuf::Descriptor* descriptor = message.GetDescriptor();
+    const Protobuf::Reflection* reflection = message.GetReflection();
+    for (int i = 0; i < descriptor->field_count(); ++i) {
+      const auto* field = descriptor->field(i);
+
+      // If this field is not in use, continue.
+      if ((field->is_repeated() && reflection->FieldSize(message, field) == 0) ||
+          (!field->is_repeated() && !reflection->HasField(message, field))) {
+        continue;
+      }
+
+      // If this field is deprecated, warn or throw an error.
+      if (field->options().deprecated()) {
+        std::string err = fmt::format(
+            "Using deprecated option '{}'. This configuration will be removed from Envoy soon. "
+            "Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for "
+            "details.",
+            field->name());
```

Comment:
`full_name()` instead of name should be more helpful.

Reply:
Thanks for the tip - this is going to be even more useful!